yktrace/src/hwt/mapper.rs

```diff
@@ -33,18 +33,15 @@ impl HWTMapper {
         for b in trace.iter_blocks() {
             match b {
                 Ok(block) => {
-                    let addr = block.start_vaddr() - self.phdr_offset;
-                    match labels.get(&addr) {
-                        Some(l) => {
-                            // FIXME Do the splitting at the time we load from DWARF.
-                            let data: Vec<&str> = l.split(':').collect();
-                            debug_assert!(data.len() == 3);
-                            let sym = String::from(data[1]);
-                            let bb_idx = data[2].parse::<u32>().unwrap();
-                            annotrace.push(SirLoc::new(sym, bb_idx))
+                    let start_addr = block.start_vaddr() - self.phdr_offset;
+                    let end_addr = start_addr + block.len();
+                    for (addr, (sym, bb_idx)) in labels {
+                        if addr >= &start_addr && addr < &end_addr {
```

Comment:
Is `end_addr` not inclusive? It's also generally more idiomatic to write:

```rust
if *addr >= start_addr && *addr < end_addr
```

though they're functionally equivalent.

Reply:
`end_addr` is computed as `start_add + len`, so it is indeed exclusive.

Reply:
I stand corrected!

Reply:
How about `(start_addr..end_addr).contains(&addr)`?

Reply:
Isn't that going to be rather inefficient?

Reply:
Agreed. That's why I used your suggestion.

Reply:
I investigated.

It's a little less efficient because `Range::contains()` uses a trait (`RangeBounds`) under the hood.

But `Range` doesn't actually allocate the range (ala Python 2 `range`). It just stores the lower and upper bounds.